rasa/utils/tensorflow/rasa_layers.py

```diff
@@ -30,7 +31,165 @@
+
+        Args:
+            new_sparse_feature_sizes: sizes of current sparse features.
+            old_sparse_feature_sizes: sizes of sparse features the model was
+                                      trained on before.
+            reg_lambda: regularization constant.
+        """
+        for name, layer in self._tf_layers.items():
+            if isinstance(layer, RasaCustomLayer):
+                layer.adjust_sparse_layers_for_incremental_training(
+                    new_sparse_feature_sizes=new_sparse_feature_sizes,
+                    old_sparse_feature_sizes=old_sparse_feature_sizes,
+                    reg_lambda=reg_lambda,
+                )
+            elif isinstance(layer, layers.DenseForSparse):
+                attribute = layer.get_attribute()
+                feature_type = layer.get_feature_type()
+                if (
+                    attribute in new_sparse_feature_sizes
+                    and feature_type in new_sparse_feature_sizes[attribute]
+                ):
+                    new_feature_sizes = new_sparse_feature_sizes[attribute][
+                        feature_type
+                    ]
+                    old_feature_sizes = old_sparse_feature_sizes[attribute][
+                        feature_type
+                    ]
+                    self._check_sparse_feature_decreased_sizes(
+                        new_sparse_feature_sizes=new_feature_sizes,
+                        old_sparse_feature_sizes=old_feature_sizes,
+                    )
+                    if sum(new_feature_sizes) > sum(old_feature_sizes):
+                        new_layer = self._create_dense_for_sparse_layer(
+                            dense_layer=layer,
+                            new_sparse_feature_sizes=new_feature_sizes,
+                            old_sparse_feature_sizes=old_feature_sizes,
+                            attribute=attribute,
+                            feature_type=feature_type,
+                            reg_lambda=reg_lambda,
+                        )
+                        self._tf_layers[name] = new_layer
+
+    @staticmethod
+    def _check_sparse_feature_decreased_sizes(
+        new_sparse_feature_sizes: List[int], old_sparse_feature_sizes: List[int],
+    ) -> None:
+        """Checks if the sizes of sparse features are decreased during fine-tuning.
+
+        Sparse feature sizes might decrease after changing the nlu file. This can
+        happen for example with `LexicalSyntacticFeaturizer`. We don't support
+        this behaviour and we raise a warning if this happens.
+
+        Args:
+            new_sparse_feature_sizes: sizes of current sparse features for a
+                                      specific attribute and feature type.
+            old_sparse_feature_sizes: sizes of sparse features the model was trained on
+                                      before for a specific attribute and feature type.
+        """
```

Comment:
The docstring should use `Raises` section to mention anything about the exceptions raised by the function.
https://www.notion.so/rasa/Python-Code-Conventions-26835ffd72484dd699f28fa693916d2e#e355fcefe85f452bb687af72752b8fc4